### DISCUSSION TITLE
Stuck in status "Starting z-wave driver"

### DISCUSSION THREAD

**Discussion by vegartb** [zwave-js/node-red-contrib-zwave-js#266] [Q&A]
I have bought the z-pi 7 from Aeotech. I Have followed their instructions and disabled the bluetooth. Then the serialport should be ttyAMA0. When I Select this serialport in zwave-js the controller node don`t work. I can not find the solution. Does anyone have a tip?? :-)
![Skjermbilde 2023-01-02 193821](https://user-images.githubusercontent.com/31689505/210271131-c3bbef86-beaf-45c9-9abe-d7fb1c5d0068.jpg)
![Skjermbilde 2023-01-02 194217](https://user-images.githubusercontent.com/31689505/210271142-9ff12ffb-9014-428d-ac21-bf2694c02dd8.jpg)

**Comment by marcus-j-davies** [comment]
Hi @vegartb 
what serial ports do you see listed in the drop down?

EDIT
Not to worry I can see them listed 😅 

Have you followed this fully?
https://aeotec.freshdesk.com/support/solutions/articles/6000230551-z-pi-7-user-guide-

There should be no reason it will not be detected by zwave js

is anything appearing in the debug console of Node RED
if after the reboot its not responding, I would think that the Bluetooth stack is still ocupying `ttyAMA0`

**Reply by marcus-j-davies** [reply]
It might Evan be worth power cycling the pi for a few seconds.

**Reply by vegartb** [reply]
I have tried a complete reboot. The bluetooth is apparently disabled, the icon is gone in the top-right corner of the pi`s desktop.

In the drop down menu in the controller node there are two options:
![skjermdump3](https://user-images.githubusercontent.com/31689505/210275724-22d3e200-fa06-419a-8473-ab0e31be07d5.jpg)
This makes no difference...
I believe I have followed the Aeotech Quick start. The only error source I see is the order of the code in the config.txt. I have put the two lines(dtoverlay=disable-bt
enable_uart=1)
I have placed them in the topp (line1):
![Skjermbilde 2023-01-02 213918](https://user-images.githubusercontent.com/31689505/210276480-ca4cb622-5145-46bc-a7fa-3309823533bf.jpg)
...I have also tried putting them at the bottom with the same result :-(
vegar

**Reply by vegartb** [reply]
The debugwindow:
![Skjermbilde 2023-01-02 214426](https://user-images.githubusercontent.com/31689505/210276798-4c8ee01a-9fb0-4aed-a1fd-5f55fc74dd5f.jpg)
...this gives a clue... or not? :-( :-)

**Reply by vegartb** [reply]
...Sorry, this was with the wrong port selected. This is the ttyAMA0:
![Skjermbilde 2023-01-02 214811](https://user-images.githubusercontent.com/31689505/210277088-3297f766-9995-4a9a-9f2b-2d47767031c7.jpg)
...but also this makes no sense to me :-)

**Reply by marcus-j-davies** [reply]
Ok this makes more sense.
What user is running your Node RED instance? you need to give serial permission to this user.

try this
`sudo usermod -a -G dialout yourusername`

change `yourusername` it must be the user that is running Node RED

then try and reboot

The error you are seeing is just a permissions problem.
this is almost excepted as this device is usually only for bluetooth that Node RED wouldn't normally have access to.

**Reply by vegartb** [reply]
Hi :-)
the Node-red is installed in root...
Tried usermod on user pi (default user on boot).... reboot... made no difference. tried the same  and root user, stil permission denied. 

This is a Pi4 with new image (less then one month). therfore pi as default user :-)

**Reply by marcus-j-davies** [reply]
so Node RED red is running **as** root or **in** root?

**Reply by marcus-j-davies** [reply]
run this in an exec node `whoami` to find out the user that Node Red is running as

**Reply by marcus-j-davies** [reply]
Also try adding pi to this group also

`sudo usermod -a -G tty pi`

so all together 
```
sudo usermod -a -G dialout pi
sudo usermod -a -G tty pi
```

**Reply by vegartb** [reply]
whoami:
![Skjermbilde 2023-01-02 224131](https://user-images.githubusercontent.com/31689505/210279764-96f21686-72f9-49d9-ba03-969f0d06b67d.jpg)

I did both the usermod, but still the same permission denied... :-P

**Reply by marcus-j-davies** [reply]
ok do this.
`cat /dev/ttyAMA0`

do you get access denied?

and you restarted after adding to both groups?

**Reply by vegartb** [reply]
sudo reboot. yepp.
This is after boot:
![Skjermbilde 2023-01-02 225248](https://user-images.githubusercontent.com/31689505/210280336-75805c0b-469f-4aac-bd9f-ec15b63809aa.jpg)
...I also tried usermod as root user (sudo -s)...
No access.

**Reply by marcus-j-davies** [reply]
Ok,

What does this return.

`ls -l /dev/ttyAMA0`

**Comment by vegartb** [comment]
![Skjermbilde 2023-01-03 063040](https://user-images.githubusercontent.com/31689505/210304446-6bce6265-9bac-464c-9e78-f6b3b685ed43.jpg)

**Reply by marcus-j-davies** [reply]
Ok,

quick test

run this  -> `id`, to make sure the `pi` user is apart of `tty`

if so run this.
`sudo chmod 660 /dev/ttyAMA0` (don't restart) but instead restart node red.

This adds read/write to this device for the group `tty`

For what its worth, all that is happening here, is the device `/dev/ttyAMA0` cannot be accessed by the `pi` user (that node red is running under)

**Reply by vegartb** [reply]
The chmod 660 was not allowed as pi user:
![Skjermbilde 2023-01-03 164711](https://user-images.githubusercontent.com/31689505/210392181-9cacd5af-0e1f-4a4c-a82a-044f28a951bb.jpg)
When I run the command as root the error msg. change in node-red....
![Skjermbilde 2023-01-03 165252](https://user-images.githubusercontent.com/31689505/210393161-a5a65eff-85a4-4408-9ab0-49832730f121.jpg)

**Reply by vegartb** [reply]
ok.... this is not easy to understand... You where right about the conflikt on the port....
But it did not resolve befor I turned off serial console:
![Skjermbilde 2023-01-03 172036](https://user-images.githubusercontent.com/31689505/210397695-0b4a0806-a74a-4196-8302-c5160dff4868.jpg)
![Skjermbilde 2023-01-03 171643](https://user-images.githubusercontent.com/31689505/210397726-76cd5987-b213-4fab-b071-15c9e105cac7.jpg)

Sorry fore all the hasl!! I will now try to test the rest of the nodes :-) Thanks for all help!! I am still not sure about what did the trick... should probably have turned off everything first? I could reinstall everything and try once more... :-)

**Reply by marcus-j-davies** [reply]
No need to apologies.

Aeotec need to update their guide 😒 
glad your up and running
